### ISSUE TITLE
Unit Tests fails on Windows without Admin privs

### ISSUE THREAD

**Issue by lowell80** [Kintyre/ksconf#20]
Unit tests fail due to permission issues on Windows unless run in "Administrator" mode, which shouldn't be necessary.

Multiple tests fail, here one example:

```
======================================================================
ERROR: test_read_file (test_parser.ParserTestCase)
Confirm that parse_conf() works with an OS-level file.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\Lowell\Sandbox\ksconf\tests\test_parser.py", line 46, in test_read_file
    tfile = os.tmpfile()
OSError: [Errno 13] Permission denied
```

**Comment by lowell80** [comment]
Problem appears to be related to the use of `os.tempfile()`.  In several, but not all cases `StringIO` could be used instead to completely avoid writing to a file system object.

**Comment by lowell80** [comment]
Switched to using StringIO() as many places a possible.  Fixed as of v0.5.0 (ebfcf0e888d49002)
